references/cli/addon.go

```diff
@@ -428,6 +428,70 @@ If you don't want the Helm component, just omit the three Chart-related paramete
 	return cmd
 }
 
+// NewAddonPushCommand pushes an addon dir/package to a ChartMuseum
+func NewAddonPushCommand(c common.Args) *cobra.Command {
+	p := &pkgaddon.PushCmd{}
+	cmd := &cobra.Command{
+		Use:   "push",
+		Short: "uploads an addon package to ChartMuseum",
+		Long: `Uploads an addon package to ChartMuseum.
+
+Two arguments are needed <addon directory/package> and <name/URL of ChartMuseum>.
+
+The first argument <addon directory/package> can be:
+	- your conventional addon directory (containing metadata.yaml). We will package it for you.
+	- packaged addon (.tgz) generated by 'vela addon package' command
+
+The second argument <name/URL of ChartMuseum> can be:
+	- registry name (helm type). You can add your ChartMuseum registry using 'vela addon registry add'.
+	- ChartMuseum URL, e.g. http://localhost:8080`,
+		Example: `# Push the addon in directory <your-addon> to a ChartMuseum registry named <localcm>
+$ vela addon push your-addon localcm
+
+# Push packaged addon mongo-1.0.0.tgz to a ChartMuseum registry at http://localhost:8080
+$ vela addon push mongo-1.0.0.tgz http://localhost:8080
+
+# Force push, overwriting existing ones
+$ vela addon push your-addon localcm -f`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 2 {
+				return fmt.Errorf("two arguments are needed: addon directory/package, name/URL of Chart repository")
+			}
+
+			c, err := c.GetClient()
+			if err != nil {
+				return err
+			}
+
+			p.Client = c
+			p.Out = cmd.OutOrStdout()
+			p.ChartName = args[0]
+			p.RepoName = args[1]
+			p.SetFieldsFromEnv()
+
+			return p.Push(context.Background())
+		},
+	}
+
+	f := cmd.Flags()
+	f.StringVarP(&p.ChartVersion, "version", "v", "", "Override chart version pre-push")
+	f.StringVarP(&p.AppVersion, "app-version", "a", "", "Override app version pre-push")
+	f.StringVarP(&p.Username, "username", "u", "", "Override HTTP basic auth username [$HELM_REPO_USERNAME]")
+	f.StringVarP(&p.Password, "password", "p", "", "Override HTTP basic auth password [$HELM_REPO_PASSWORD]")
+	f.StringVarP(&p.AccessToken, "access-token", "", "", "Send token in Authorization header [$HELM_REPO_ACCESS_TOKEN]")
+	f.StringVarP(&p.AuthHeader, "auth-header", "", "", "Alternative header to use for token auth [$HELM_REPO_AUTH_HEADER]")
+	f.StringVarP(&p.ContextPath, "context-path", "", "", "ChartMuseum context path [$HELM_REPO_CONTEXT_PATH]")
+	f.StringVarP(&p.CaFile, "ca-file", "", "", "Verify certificates of HTTPS-enabled servers using this CA bundle [$HELM_REPO_CA_FILE]")
+	f.StringVarP(&p.CertFile, "cert-file", "", "", "Identify HTTPS client using this SSL certificate file [$HELM_REPO_CERT_FILE]")
+	f.StringVarP(&p.KeyFile, "key-file", "", "", "Identify HTTPS client using this SSL key file [$HELM_REPO_KEY_FILE]")
```

Comment:
by default, users won't set the certifcates, would that work? So this is not required, right?

Reply:
Yes, they are only required if the user uses a self-signed certificate.

Reply:
what if users not using the self-signed certificate?

Reply:
If that's the case, they do not need to do anything. Just push it and it just works.

Reply:
ok